### DISCUSSION TITLE
Integrated file manager

### DISCUSSION THREAD

**Discussion by peaklabs-dev** [coollabsio/coolify#3141] [Feature Requests]
## Idea
Add a full-featured file manager to coolify. 
- 1 file manager in the settings tab where we can see and manage all files of the coolify instance and the reverse proxy (server file manger). 
- And a file manager at the service/application level, where we can easily manage all files of a container or service in a clean UI interface.

## UI inspiration
![image](https://github.com/coollabsio/coolify/assets/122374094/4d310f62-4fd4-441f-b716-235839c1ef33)

**Comment by Emahhh** [comment]
It would be a nice feature to have. 
Maybe it would be easier to integrate something like https://github.com/prasathmani/tinyfilemanager

**Comment by sianami123** [comment]
i want edit wp-config.php in my wordpress site
how can i edit this file without file manager?

**Reply by samuelh2005** [reply]
Can't you edit the file through the terminal?

**Reply by pro-target** [reply]
In principle, you can manage the server from the terminal.

Without Coolify.

**Reply by joe-sal** [reply]
you can use sftp with a tool like filezilla or cyberduck, some desktop editors has also the function to connect with sftp

**Reply by faizananwerali** [reply]
> Can't you edit the file through the terminal?

Well, I can't tell my clients to use the terminal to edit files or upload their data files.

**Comment by ChiragDhawan630** [comment]
Coolify doesn't even come with any file manager thet gives us access to docker volumes. We have to use File Browser that can't even add files, adding files saves them with root permissions, these permissions can't be changed from UI so web apps can't use them due to permission issues !!!

**Comment by TriHydera** [comment]
Also would be handy for a ftp or webdav container to upload files
